app/models/concerns/orchestration/external_ipam.rb

```diff
@@ -0,0 +1,146 @@
+module Orchestration::ExternalIPAM
+  extend ActiveSupport::Concern
+  include Orchestration::Common
+  include SubnetsHelper
+
+  included do
+    after_validation :queue_external_ipam
+    before_destroy :queue_external_ipam_destroy
+  end
+
+  def generate_external_ipam_task_id(action, network_type, interface = self)
+    id = [interface.mac, interface.ip, interface.identifier, interface.id].find { |x| x&.present? }
+    "external_ipam_#{action}_#{id}_#{network_type}"
+  end
+
+  protected
+
+  def set_add_external_ip(params)
+    ip, subnet = params[:ip], params[:subnet]
+
+    if ip_is_available?(ip, subnet)
+      subnet.external_ipam_proxy.add_ip_to_subnet(ip, subnet.network_address, subnet.externalipam_group)
+    else
+      self.errors.add :ip, _('This IP address has already been reserved in External IPAM') if subnet.network_type == "IPv4"
+      self.errors.add :ip6, _('This IP address has already been reserved in External IPAM') if subnet.network_type == "IPv6"
+      self.errors.add :interfaces, _('Some interfaces are invalid')
+    end
+  end
+
+  def set_remove_external_ip(params)
+    ip, subnet = params[:ip], params[:subnet]
+    subnet.external_ipam_proxy.delete_ip_from_subnet(ip, subnet.network_address, subnet.externalipam_group)
+  end
+
+  def del_add_external_ip(params)
+    new_record? ? rollback_on_add : rollback_on_update
+  end
+
+  def del_remove_external_ip(params)
+  end
```

Comment:
I take this that transaction compensation is not possible in this case as IPAM will return a different IP address. Therefore I'd put a warning into Rails log explaining that deletion failed and it cannot be compensated.